_posts/2018-01-03-Continuous_delivery_with_episerver_dxcs.md

```diff
@@ -0,0 +1,459 @@
+
+Continuous integration:
+* Make code available to other developers
+* Make sure code builds after merging to works of other people 
+* Track the quality of your code 
+* Test your code 
+
+Continuous delivery:
+* Make code available to your end-users 
+* Make sure that you can deploy any time 
+* Track the quality of your runtime 
+* Test your deployments 
+
+I feel that continuous integration is something that software business is doing already with high adaptation rate. Continuous delivery in the other hand is something that many are still learning. Continuous delivery definitely needs more maturity of the development team since you need to automate a lot of things to be able to say to your CI ecosystem to deploy automatically. You will need a process that makes all kind of sanity checks across the line to trust the deployments on the hand of the automated system. The good thing is once you get on the road it will be a lot easier to adjust the process when you have ability to repeat it multiple times a day. 
+
+The hardest question asked about continuous delivery is most often the following: "How can your customer decide when a feature is good enough for release?". This is something that you need to decide per feature. With Episerver creating new content features is easy. Customer tests the new content types and decides when they serve them well enough to be published on the live. Changing existing content can be done with feature toggles. Of course creating a feature toggle is a change too and can be errornous. Here comes the place the need for automatic testing. You need to catch the errors before they get on the production. The hardest kind of changes are cross-cutting concerns like authentication or database schema changes. You need to be super careful how do you implement and roll this kind of features on the production. In the end the most important thing is that your development organization (developers and business people) have the correct working habits and mindset for failing fast and failing often. Some day you will fail and with CD you will be most likely to recover faster. 
+
+Having said scary things about deploying features there is the positive side of the coin. When you deploy often you have only small changeset at the time. It is a lot easier to find the problems on your patch and fix them. Installing the fix is also easy when you have process to take care of all and when no manual work is not involved. 
```

Comment:
a small changeset
when manual work is not involved tai when no manual work is involved